implementation/Implementation.py

```diff
@@ -49,12 +44,27 @@ def fup(self):
             sumPP += LA.norm(pArray[p]) ** 2
         foo = self.lamb * (sumUU + sumPP)
         fup = sumRUP + foo
-        print(fup)
+        # print(fup)
+
+    def calculatingErrors(self, r):
+        sumErrors = 0
+        qErrors = 1
+        for i in range(0, len(self.rCopy)):
+            for j in range(0, len(self.rCopy[0])):
+                if (self.rCopy[i][j] != 0):
+                    sumErrors += abs(self.rCopy[i][j] - r[i][j])
+                    qErrors+=1
+        aproxErrors = sumErrors/qErrors
+        return aproxErrors
 
-    #
+    ##################################
 
     def do_alg(self):
         self.r, id_list = self.parser.getparsed()
+        self.rCopy = [[0 for i in range(0, len(self.r[0]))] for j in range(0, len(self.r))]
```

Comment:
Ja chyba użyłem raz numpy.copy() gdzieś i tez mi działało, zobacz, czy tu tez będzie ok

Reply:
taa, działa, zapomniałam to cofnąć jak co innego naprawiałam